mongoDbTest.js

```diff
@@ -0,0 +1,281 @@
+			MongoClient.connect(url, {
+				useNewUrlParser: true,
+				useUnifiedTopology: true,
+			}),
+			new Promise((res,rej) => {
+				setTimeout(() => rej("Timed out after 15 seconds."), 15000)
+			})
+		]);
+		db = mongo.db(dbName);
+		console.log(`Connected to "${url}", using database "${argv.dbName}"`);
+	} catch (error) {
+		console.log(); // Newline!
+		console.error(`Tried connecting to ${url}, using database ${dbName}`);
+		console.error("Ran into an error while connecting to MongoDB!");
+		console.error('Error object:');
+		console.dir(error);
+		console.error('Exiting early due to errors!')
+		// TODO: consider any cleanup code before exiting: any open file handles?
+		process.exit();
+	} finally {
+		clearInterval(dotAnim);
+	}
+	return [mongo, db];
+}
+
+console.log("Connecting to MongoDB instance…");
+try {
+	[mongo, db] = await connectMongo(url, dbName);
+} catch (err) {
+	console.error(`Could not connect to MongoDB instance at ${url}!`);
+}
+
+
+
+app.use(express.json());
+app.use(session(
+	{
+	  secret: "shhhh...secret",
+	  name: "ShakeGuardSessionID",
+	  resave: false,
+	  // create a unique identifier for that client
+	  saveUninitialized: true
+	})
+  );
+// static path mappings
+app.use("/js", express.static("public/js"));
+app.use("/css", express.static("public/css"));
+app.use("/images", express.static("public/images"));
+app.use("/html", express.static("public/html"));
+
+app.get('/', function (req, res) {
+	let doc = fs.readFileSync("./html/mongo.html", "utf8");
+	res.send(doc);
+});
+
+// Create example 
+app.post('/create-person', async (req, res) => {
+	console.log(req.body);
+	const name = req.body.name;
+	const email = req.body.email;
+
+	// Set response header regardless of success/failure
+	res.setHeader("Content-Type", "application/json");
+	try {
+		const insertResult = await db.collection('testCollection').insertOne({name, email});
+		console.log(insertResult);
+
+		// Send success response
+		res.send({ status: "success", msg: "successfully added person." });
+	} catch(e) {
+		console.log(e);
+		//send fail response
+		res.send({ status: "fail", msg: "could not add person." });
+	} 
+})
+
+// Read example
+// This example just gets all the documents in a collection
+// use .fineOne() instead to get just one document
+app.get('/people', async function (req, res) {
+	// Get every document in 'testCollection' collection
+	// send names to client
+  	const results = await db.collection('testCollection').find({}).toArray();
+  	const names = results.map(element => {
+		return element.name;
+	})
+
+	// Send query result
+	res.send(names);
+})
+
+// Update example
+app.post('/change-name', async (req, res) => {
+	const givenEmail = req.body.email;
+	const newName = req.body.name;
+
+	// Set response header regardless of success/failure
+	res.setHeader("Content-Type", "application/json");
+	try {
+		const results = await db.collection('testCollection').updateOne({email: givenEmail}, {$set: {name: newName}})
```

Comment:
Is `$set` a convention in the MongoDB library, or a property name you came up with yourself? Should probably read more about the Node Mongo lib… 😅

Reply:
It just tells MongoDB to set the field `name` in this instance to the value of `newName`

https://www.mongodb.com/docs/manual/reference/operator/update/set/